### PR TITLE
Add inspect command for proxy_ocr2

### PR DESCRIPTION
**Pull request by dimriou**

### CODE REVIEW

File: packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  aggregator: string
+  version: string
+  description: string
+}
+
+type ContractExpectedInfo = {
+  aggregator: string
+  version: string
+  description: string
+  phase_id?: number
```
Comment (RodrigoAD) [inline packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts:18]: Nit picking here, let's make everything camelCase

File: packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+const instruction: InspectInstruction<CommandInput, ContractExpectedInfo> = {
+  command: {
+    category: CATEGORIES.PROXIES,
+    contract: CONTRACT_LIST.PROXY_OCR_2,
+    id: 'inspect',
+    examples: ['proxy_ocr2:inspect --network=<NETWORK> <CONTRACT_ADDRESS>'],
+  },
+  instructions: [
```
Comment (RodrigoAD) [inline packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts:112]: Let's add `latest_round_data` instruction too, and simply print its details

**Review by RodrigoAD — COMMENTED** [review]
Almost good to go, some small comments

File: packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+const inspect = (expected: ContractExpectedInfo, onchainData: ContractExpectedInfo): boolean => {
+  let inspections: inspection.Inspection[] = [
+    inspection.makeInspection(onchainData.aggregator, expected.aggregator, 'Aggregator'),
+    inspection.makeInspection(onchainData.version, expected.version, 'Version'),
+    inspection.makeInspection(onchainData.description, expected.description, 'Description'),
+  ]
+
+  logger.line()
+  logger.info('Inspection results:')
+  logger.info(`Aggregator:
```
Comment (RodrigoAD) [inline packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts:78]: We can put all this info into a single log:
```js
logger.info(`
 - Aggregator: ${onchainData.aggregator}
 - Version: ${onchainData.version}
 ...
`)
```

File: packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  logger.info(`Decimals: 
+    - Decimals: ${onchainData.decimals}
+  `)
+
+  logger.info(`Owner: 
+    - Owner: ${onchainData.owner}
+  `)
+
+  logger.info(`Last Round Data:
+    - Data: ${JSON.stringify(onchainData.latestRoundData)}
```
Comment (RodrigoAD) [inline packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts:103]: Let's print the exact details here:
```js
  if (onchainData.latestRoundData) {
    logger.info(`Last Round Data:
    - Round ID: ${onchainData.latestRoundData?.roundId}
    - Answer: ${Number(onchainData.latestRoundData?.answer) / 10 ** Number(onchainData.decimals)}
    - Transmission at: ${new Date(onchainData.latestRoundData.transmissionTimestamp * 1000)}
    - Observation at: ${new Date(onchainData.latestRoundData.observationsTimestamp * 1000)}
  `)
  } else {
    logger.warn('Latest round not found')
  }
```
Notice that the timestamp comes in Unix format. I'd recommend to have a `dateFromUnix` helper function into `/lib/utils.ts` as we've been seeing it for some time now. If you do, please add a "TODO" to move that to `gauntlet-core` at some point

File: packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+  logger.info(`Decimals: 
+    - Decimals: ${onchainData.decimals}
+  `)
+
+  logger.info(`Owner: 
+    - Owner: ${onchainData.owner}
+  `)
+
+  logger.info(`Last Round Data:
```
Comment (RodrigoAD) [inline packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts:102]:
```suggestion
  logger.info(`Latest Round Data:
```

File: packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+const makeInput = async (flags: any, args: string[]): Promise<CommandInput> => {
+  if (flags.input) return flags.input as CommandInput
+  const rdd = RDD.getRDD(flags.rdd)
+  const contract = args[0]
+  const info = rdd.proxies[contract]
+
+  return {
+    aggregator: info.aggregator,
+    version: info.contractVersion,
+    description: info.name,
```
[the remark this answers is not in the document]
Comment (dimriou) [reply packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts:33]: We had a quick talk about this during standup:  `decimals` and `owner` are also included in _ocr2:inspect_ command so they were excluded from the inspection part. Should I still include them? Indeed I can associate proxy with ocr2 contract through the RDD and compare them.

File: packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+type ContractExpectedInfo = {
+  aggregator: string
+  version: string
+  description: string
+  phaseId?: number
+  decimals?: string | number
+  owner?: string
+  latestRoundData?: object
```
[the remark this answers is not in the document]
Comment (dimriou) [reply packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts:21]: Good call. Maybe we could transfer these general-purpose types to a more suited place if they start to increase in number.

File: packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (RodrigoAD) [reply packages-ts/gauntlet-terra-contracts/src/commands/contracts/proxy_ocr2/inspection/inspect.ts:33]: Let's add it yeah

**Review by RodrigoAD — APPROVED** [review]
Awesome, thanks @dimriou . Feel free to merge

**Review by RodrigoAD — APPROVED** [review]

**Pull request merged**